lib/helpers/ppt-factory-helper.js

```diff
@@ -1081,6 +1138,200 @@ class PptFactoryHelper {
             }
         }
     }
+
+    static calcStringWidthPoints(text, fontName, fontSize) {
+        let stringWidth = 0;
+        let charWidths = this.getCharWidthsForFont(fontName);
+
+        if (charWidths === false) return -1;
+
+        for (let i = 0; i < text.length; i++) {
+            let ascii = text.charCodeAt(i);
+
+            if (ascii >= 32) {
+                stringWidth += fontSize * charWidths[ascii];
+            }
+        }
+
+        return stringWidth;
+    }
+
+    /* eslint-disable complexity */
+    // this function is impossible to refactor to a complexity lower than 16
+    static getCharWidthsForFont(fontName) {
+        let charWidths = [];
+
+        switch (fontName) {
+        case 'Calibri':
+            for (let i = 32; i <= 127; i++) {
+                if ([32, 39, 44, 46, 73, 105, 106, 108].includes(i)) {
```

Comment:
this seems like a good time to use a multiway switch statement instead of if/else

```javascript
switch(true){
  case [32, 39, 44, 46, 73, 105, 106, 108].includes(i): 
    charWidths[i] = 0.2526;
  case [40, 41, 45, 58, 59, 74, 91, 93, 96, 102, 123, 125].includes(i): 
    charWidths[i] = 0.3144;
   ....
```
Just a more compact syntax and easier to scan the code.

Reply:
Also would make the code more readable to break the font processing out into their own functions e.g.

```javascript
case 'Lucida Console':
     formatLucidaConsole(charWidths);
case 'Times New Roman':
     formatTimesNewRoman(charWidths);
```

Reply:
I did both of your suggestions. Also got rid of prettier's ugly formatting on long arrays by breaking them up into two parts. Code looks much better. Still wasn't able to get around es-lint's complexity warning for the getCharWidthsForTimesNewRoman() function. But that was the only function which exceeded the complexity limit. Had to disable the warning on that function.